internal/services/network/subnet_resource.go

```diff
@@ -25,39 +25,64 @@ import (
 var SubnetResourceName = "azurerm_subnet"
 
 var subnetDelegationServiceNames = []string{
+	"GitHub.Network/networkSettings",
 	"Microsoft.ApiManagement/service",
+	"Microsoft.Apollo/npu",
 	"Microsoft.App/environments",
+	"Microsoft.App/testClients",
+	"Microsoft.AVS/PrivateClouds",
 	"Microsoft.AzureCosmosDB/clusters",
 	"Microsoft.BareMetal/AzureHostedService",
+	"Microsoft.BareMetal/AzureHPC",
+	"Microsoft.BareMetal/AzurePaymentHSM",
 	"Microsoft.BareMetal/AzureVMware",
 	"Microsoft.BareMetal/CrayServers",
+	"Microsoft.BareMetal/MonitoringServers",
 	"Microsoft.Batch/batchAccounts",
+	"Microsoft.CloudTest/hostedpools",
+	"Microsoft.CloudTest/images",
+	"Microsoft.CloudTest/pools",
+	"Microsoft.Codespaces/plans",
 	"Microsoft.ContainerInstance/containerGroups",
 	"Microsoft.ContainerService/managedClusters",
+	"Microsoft.ContainerService/TestClients",
 	"Microsoft.Databricks/workspaces",
 	"Microsoft.DBforMySQL/flexibleServers",
+	"Microsoft.DBforMySQL/servers",
 	"Microsoft.DBforMySQL/serversv2",
 	"Microsoft.DBforPostgreSQL/flexibleServers",
 	"Microsoft.DBforPostgreSQL/serversv2",
 	"Microsoft.DBforPostgreSQL/singleServers",
+	"Microsoft.DelegatedNetwork/controller",
+	"Microsoft.DevCenter/networkConnection",
+	"Microsoft.DocumentDB/cassandraClusters",
+	"Microsoft.Fidalgo/networkSettings",
 	"Microsoft.HardwareSecurityModules/dedicatedHSMs",
 	"Microsoft.Kusto/clusters",
-	"Microsoft.Logic/integrationServiceEnvironments",
 	"Microsoft.LabServices/labplans",
+	"Microsoft.Logic/integrationServiceEnvironments",
 	"Microsoft.MachineLearningServices/workspaces",
 	"Microsoft.Netapp/volumes",
 	"Microsoft.Network/dnsResolvers",
-	"Microsoft.Network/managedResolvers",
```

Comment:
should we be removing this value? it could break someones config so we should keep it around to remove in 4.0 if it is no longer valid